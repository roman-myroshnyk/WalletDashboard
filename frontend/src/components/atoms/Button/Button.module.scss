.button {
  width: max-content;
  padding: 15px 25px;

  color: black;
  font-weight: bold;
  font-size: 16px;
  letter-spacing: -0.2px;
  text-align: center;
  word-break: keep-all;
  overflow-wrap: break-word;

  border: none;
  border-radius: 8px;
  outline: none;

  resize: none;
  user-select: none;
  -webkit-text-stroke: 0 black;
  -webkit-text-fill-color: white;
  -webkit-tap-highlight-color: transparent;

  -webkit-touch-callout: none;
}

.button:disabled {
  pointer-events: none;
}
