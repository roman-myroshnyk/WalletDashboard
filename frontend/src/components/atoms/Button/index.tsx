import { FC, MouseEventHandler } from 'react';
import { motion } from 'framer-motion';
import { animations, transition } from '@/animations/ButtonAnimations';
import styles from './Button.module.scss';

export interface IProps {
    text: string
    disabled?: boolean
    onClick?: MouseEventHandler<HTMLButtonElement>
}

const Button:FC<IProps> = ({ text, disabled, onClick }) => (
  <motion.button
    className={styles.button}
    type="button"
    transition={transition}
    initial="default"
    animate="default"
    variants={animations}
    disabled={disabled}
    whileHover="hover"
    whileTap="pressed"
    {...(!disabled && {
      onClick,
    })}
  >
    {text}
  </motion.button>
);

Button.defaultProps = {
  disabled: false,
  onClick: undefined,
};

export default Button;
