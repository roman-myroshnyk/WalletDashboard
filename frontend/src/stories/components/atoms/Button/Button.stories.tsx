import { Meta, Story } from '@storybook/react/types-6-0';
import Button, { IProps } from '@/atoms/Button';

export default {
  title: 'atoms/Button',
  component: Button,
  argTypes: {
    text: {
      controls: 'text',
    },
    disabled: {
      controls: 'boolean',
    },
    onClick: {
      action: 'onClick: clicked',
    },
  },
} as Meta;

const Template:Story<IProps> = (args) => (
  <Button {...args} />
);

export const Default = Template.bind({});

Default.args = {
  text: 'Check',
  disabled: false,
};
