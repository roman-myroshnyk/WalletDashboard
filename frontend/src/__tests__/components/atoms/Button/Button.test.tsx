import { fireEvent, render } from '@testing-library/react';
import Button from '@/atoms/Button';

describe('atoms/Button', () => {
  it('renders Button in normal state', () => {
    const { container } = render(<Button text="I am button" disabled={false} />);
    expect(container).toMatchSnapshot();
  });

  it('renders Button in disabled state', () => {
    const { container } = render(<Button text="I am button" disabled />);
    expect(container).toMatchSnapshot();
  });

  it('Button is enabled, onClick should fire', () => {
    const handleClick = jest.fn();
    const { getByText } = render(<Button
      text="I am button"
      disabled={false}
      onClick={handleClick}
    />);
    const button = getByText('I am button');
    fireEvent.click(button);
    expect(handleClick).toBeCalledTimes(1);
  });

  it('Button is disabled, onClick should not fire', () => {
    const handleClick = jest.fn();
    const { getByText } = render(<Button
      text="I am button"
      disabled
      onClick={handleClick}
    />);
    const button = getByText('I am button');
    fireEvent.click(button);
    expect(handleClick).toBeCalledTimes(0);
  });
});
