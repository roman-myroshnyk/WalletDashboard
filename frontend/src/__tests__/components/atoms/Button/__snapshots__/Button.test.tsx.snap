// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Button renders Button in disabled state 1`] = `
<div>
  <button
    class="button"
    data-testid="button"
    disabled=""
    style="transform: none; background: rgb(0, 204, 255);"
    type="button"
  >
    I am button
  </button>
</div>
`;

exports[`atoms/Button renders Button in normal state 1`] = `
<div>
  <button
    class="button"
    data-testid="button"
    style="transform: none; background: rgb(0, 204, 255);"
    type="button"
  >
    I am button
  </button>
</div>
`;
